### DISCUSSION TITLE
[GUIDE] Switching From Proprietary NVIDIA Drivers to NVK

### DISCUSSION THREAD

**Discussion by liamrahum** [hyprwm/Hyprland#5633] [General]
So I've been using Hyprland+NVIDIA for over 6 months now, and ever since the shiny new Vulkan drivers came out I've been wanting to use them instead of my proprietary drivers (I don't really use my GPU intensely so it's okay for me) - but I've noticed that even while all the other DEs and WMs look smooth with NVK - Hyprland still suffered from stutters and lags, and I **finally** figured out why:

## First - Install NVK Drivers (Duh)
I know the drivers are in the Linux kernel, but personally I like having the latest (-git) version for best performance.
On Arch you can install them with:
``yay -S vulkan-nouveau-git lib32-vulkan-nouveau-git``

## Second - Change the Mkinitcpio config file (/etc/mkinitcpio.conf)
Remember those modules we had to put there for the proprietary drivers (nvidia nvidia_uvm.. etc)? Well, it's time to replace them with a module called ``nouveau`` - so like this:
```MODULES=(nouveau)```
#### Then execute mkinitcpio for your Kernels of choice (linux, linux-lts..):
```sudo mkinitcpio -P linux```

### This is the step that removed my stutters for Hyprland
Remember, you're not a proprietary driver user now, you're using Vulkan - so **comment out** the lines you used for the proprietary drivers:
```
# COMMENT THESE OUT
env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = WLR_NO_HARDWARE_CURSORS,1
```

## Fourth (prob optional) - Change GRUB's config
remove the ``nvidia-drm.modeset=1`` part, then execute ``sudo grub-mkconfig -o /boot/grub/grub.cfg``
 
## Finally - Restart your computer, and enjoy the NVK drivers :)

**Comment by GokuPlay609** [comment]
Thanks

**Comment by bryanmh19** [comment]
Does Hyprland run smoothly with NVK drivers sir? Because I'm having problems with it and my monitors at this moment, so I'm looking for anything that could help…
